src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -50,39 +45,34 @@ private Profile() {
       blavf.setWeight(c, 0d);
     }
     this.linearAvf = blavf.build();
-    this.floorAreaWeightRange = null;
-    this.nbBedroomsWeightRange = null;
-    this.nbSleepingWeightRange = null;
-    this.nbBathroomsWeightRange = null;
-    this.terraceWeightRange = null;
-    this.floorAreaTerraceWeightRange = null;
-    this.wifiSubjectiveValueWeightRange = null;
-    this.pricePerNightSubjectiveValueWeightRange = null;
-    this.nbMinNightSubjectiveValueWeightRange = null;
-    this.teleSubjectiveValueWeightRange = null;
+
+    this.rangeMap = new EnumMap<>(Criterion.class);
+    for (Criterion c : Criterion.values()) {
+      this.rangeMap.put(c, null);
+    }
   }
 
   /**
-   * This function allows the user to clone an object Profile
+   * This function allows the user to clone an object Profile This function allows you to indicate
+   * which value should be cloned. Thus, each parameter not present will be cloned from the current
+   * Profile To modify the profile during the clone, just give it the desired parameters
    *
-   * @return an object Profile
+   * @param ranges corresponds to the range associated with the <code>Profile</code>. In the case
+   *     where an <code>EnumMap</code> is present, it does not have to contain all the necessary
+   *     keys
+   * @param lavf corresponds to the LinearAVF associated with the Profile. In case it is not
+   *     present, we clone the LinearAVF from current <code>Profile</code>
+   * @return Profile modified (if Optional are present)
    */
-  private Profile cloneProfile() {
-
+  private Profile clone(
+      Optional<EnumMap<Criterion, Range<Double>>> ranges, Optional<LinearAVF> lavf) {
```

Comment:
Veuillez remplacer ceci par un constructeur (EnumMap<Criterion, Range<Double>> ranges, LinearAVF lavf) (ou m’indiquer pourquoi cela ne fonctionnerait pas).

Reply:
Le but est ici de cloner certains (ou tous les) attributs.

Sauf erreur, en utilisant un constructeur, nous n'aurons pas accès à l'instance en cours (sauf si nous ajoutons un Profile dans les paramètres du constructeur). Or, pour les critères qui ne sont pas disponibles (qui ne veulent pas être changer), nous récupérons les anciens (ceux de l'instance `this`)

Reply:
Dans c509ce1eef866b504b08cf8ad8887c137837f441 la fonction `clone()` est toujours présente et un constructeur `EnumMap<Criterion, Range> ranges, LinearAVF lavf` a été ajouté.

Il pourrait être intéressant de déplacer le constructeur directement dans le `Builder`.